source/app/models/url.rb

```diff
@@ -0,0 +1,33 @@
+require 'net/http'
+
+class Url < ActiveRecord::Base
+  before_create :reset_click_count
+
+  validates :long_url, presence: true, format: { with: /http(|s):\/\/.+/ }
```

Comment:
I could nitpick this `format` regexp, but I'm not sure if it is needed. Is the `URI::regexp` matching sufficient?